### ISSUE TITLE
Alternative model of bridging

### ISSUE THREAD

**Issue by lingnand** [JojiiOfficial/Matrix-EmailBridge#30]
It seems to me that another way of modelling the bridge could be:

- each email address is represented as a matrix user
- each message thread is an email thread with the correspondent
- for emails going out to multiple addresses, it becomes a room with multiple users

This is closer to the 'simple puppeted bridging' explained in the matrix doc and in my opinion a more intuitive way of interaction for the end user.

Did you ever consider this model instead of the more bot-like model your project is using?

P.S. I'm very new to matrix so could be missing something obvious.

**Comment by JojiiOfficial** [comment]
Yes I considered this. When I started this project I was new to matrix too and wanted to learn GOlang. Using the bot sdk was easier at this time so I decided to use this instead of the appservice API. Switching to the appservice API would lead to a complete redesign of the bridge (a lot of work). I don't have time for that at the moment, maybe I'll do these huge change later.
